pkg/apis/storage/validation/validation.go

```diff
@@ -165,8 +165,33 @@ func validateAttacher(attacher string, fldPath *field.Path) field.ErrorList {
 // validateSource tests if the source is valid for VolumeAttachment.
 func validateVolumeAttachmentSource(source *storage.VolumeAttachmentSource, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if source.PersistentVolumeName == nil || len(*source.PersistentVolumeName) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath, ""))
+
+	// VolumeAttachmentSource must have either PersistentVolumeName or inlineVolumeSource
+	// but not both.
+	if source.PersistentVolumeName == nil && source.InlineVolumeSource == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, "source", "either persistentVolumeName or inlineVolumeSource must be provided"))
+	}
+
+	if source.PersistentVolumeName != nil && source.InlineVolumeSource != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, "source", "either persistentVolumeName or inlineVolumeSource can be provided"))
+	}
+
+	if (source.PersistentVolumeName != nil && len(*source.PersistentVolumeName) == 0) && source.InlineVolumeSource == nil {
```

Comment:
the check for inlinevolume source nil is already covered by the first two checks and can be removed